universal-application-tool-0.0.1/app/views/admin/programs/ProgramIndexView.java

```diff
@@ -25,30 +27,96 @@ public ProgramIndexView(AdminLayout layout) {
   }
 
   public Content render(ImmutableList<ProgramDefinition> programs) {
-    return layout.render(
-        body(
-            h1("Programs"),
-            div(
-                form(
-                    a("Add a Program")
-                        .withId("new-program")
-                        .withHref(controllers.admin.routes.AdminProgramController.newOne().url()))),
-            div(each(programs, this::shortProgram))));
+    Tag contentDiv =
+        div()
+            .withClasses(Styles.PX_20)
+            .with(
+                h1("Programs").withClasses(Styles.MY_4),
+                each(programs, this::renderProgramListItem),
+                renderNewProgramButton());
+
+    return layout.render(head(layout.tailwindStyles()), body(contentDiv));
+  }
+
+  public Tag renderNewProgramButton() {
+    return a("Create new program")
+        .withId("new-program")
+        .withHref(controllers.admin.routes.AdminProgramController.newOne().url())
+        .withClasses(
+            Styles.INLINE_BLOCK,
+            Styles.CURSOR_POINTER,
+            Styles.PY_4,
+            Styles.PX_3,
+            Styles.MY_2,
+            Styles.ROUNDED_MD,
+            Styles.RING_BLUE_200,
+            Styles.RING_OFFSET_2,
+            Styles.BG_BLUE_400,
+            Styles.TEXT_WHITE,
+            StyleUtils.applyUtilityClass(StyleUtils.HOVER, Styles.BG_BLUE_500),
+            StyleUtils.applyUtilityClass(
+                StyleUtils.FOCUS, ImmutableList.of(Styles.OUTLINE_NONE, Styles.RING_2)));
```

Comment:
Question (non-blocking): would you be up for extracting these styles to a method for creating buttons with these styles so we could start re-using?

Reply:
In addition to (rather than instead of) what Bion mentioned, another thought I had was to create other constants that are lists of classes that are useful, e.g., `CREATE_BUTTON_CLASSES`, `SUBMIT_BUTTON_CLASSES`, `EDIT_LINK_CLASSES`, etc.

I sort of attempted to do this earlier, but found it didn't play that nicely with the variable args that `withClasses` takes. Definitely not necessary to do in this PR, but just musing about potential ways to keep our style code somewhat separate from our template code so that it's easier to scan the template code...

Reply:
Yeah, I think we're going to want to reuse components as much as possible, but this is probably a good example of normalizing when it makes sense. I don't want to start pulling styles out of `ClassA` into a utility until we decide we also need it (imminently for something like buttons) in `ClassB`. Otherwise we'll get out of control in our Util or Base classes and we'll end up with single use or orphaned methods.